### PR TITLE
Feature/checkout

### PR DESCRIPTION
**Pull request by lesser1**
This is a relatively big pull request. I'm sorry about that.

First of all, [this is the Gropius issue](https://frontend.gropius.duckdns.org/components/9d12d6ac-e85a-407c-bdff-a8321fb75d3a/issues/0d686183-befe-48a6-92c3-549392d4f9bc).

Please consider that
- The user can cancel the checkout by clicking the CANCEL button or by clicking one of the buttons in the topmost app bar. The CANCEL button has a confirmation dialog -- the user has to confirm that they actually want to cancel the checkout. The buttons in the app bar do not have that. I would suggest I open an extra issue for that in order to keep this feature implementation as concise as possible.
- In the backend, the prices of the products are in Cents not in Euros. Meanwhile, as far as I know, the shipping fees are in Euros. The frontend still does not convert the product prices meaning it does not simply divide them by 100. I want to do that in a separate issue for the whole application. This means that the costs displayed during checkout are of course not correct.
- The project supervisors requested that as a user, one can provide the VAT identification number during checkout. I have yet to implement that. I would suggest that this should be part of an extra issue. Also the backend has to be extended to take care of that information.
- When testing the checkout, you need to have added a shipment method. Right now, there is no way to do that via the Frontend. Instead, you have to execute a GraphQL mutation via GraphiQL:
```
mutation MyMutation {
  createShipmentMethod(
    input: {baseFees: 10, description: "DHL kinda rules the world.", externalReference: "", feesPerItem: 10, feesPerKg: 10, name: "DHL"}
  ) {
    id
  }
}
```
Don't forget to set the authorization header:
```
{
  "authorization": "Bearer [paste your token here]"
}
```

The following issues regarding the feature implementation have already been found / tracked:
- [x] The SelectOrAddNewAddressCards and SelectOrAddNewCreditCardCard should not be as wide.
- [ ] The shipment methods are not cached, but queried anew each time when switching from address to shipment. -- into a separate issue
- [ ] Clicking on the product in the ShipmentInformationView navigates directly to the product without a "Confirm cancellation" prompt. -- into a separate issue
- [x] Explanations of what Invoice and Prepayment are would be helpful.
- [x] While the order is being created, there should be an animation indicating to the user that something is happening in the background.

### Definition of Done
- [x] Requirements of the issue are met
- [x] Code has been reviewed
- [x] Code is documented
- [x] GraphQL related artefacts are documented

### CODE REVIEW

File: src/layouts/checkout/TheCheckoutLayout.vue [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+ * Additionally "clears" or resets the order information of the app store to undefined.
+ * @returns - A promise that resolves after the order is created and the navigation to the order summary page is completed.
+ */
+async function createOrderAndNavigateToOrderSummary(): Promise<void> {
+    if (currentUserId.value == undefined) {
+        return
+    }
+    isAwaitingOrderCreation.value = true
+    try {
+        var idOfCreatedOrder = await createOrder(
```
Comment (legendofa) [inline src/layouts/checkout/TheCheckoutLayout.vue:299]: The order also needs to be placed via the GraphQL endpoint here, the idea was to create the order before showing the summary and placing it after the user has used the confirm button on the summary page

**Review by legendofa — CHANGES_REQUESTED** [review]
I tested the process of creating an order and it looks fine/works as i expected. Just one thing is missing: Invoking the endpoint to place the order

File: src/layouts/checkout/TheCheckoutLayout.vue (hunk 1 above)
Comment (lesser1) [reply src/layouts/checkout/TheCheckoutLayout.vue:299]: Ah, okay, that explains a lot! I will change it.

**Review by legendofa — APPROVED** [review]
The order placement logic works now.

However, it would be nice to see an overview of the order before pressing the placement button. I can understand if this is not possible due to time constraints, so i will approve it for now. I think the user would expect a redirect to a simple orders overview page, which shows all orders in a list. And (maybe) the same order overview page component as before placing the order + the deliver status. 

Overall i like the look and feel how the order process is now implemented. Good job!

**Comment by lesser1** [comment]
> The order placement logic works now.
> 
> However, it would be nice to see an overview of the order before pressing the placement button. I can understand if this is not possible due to time constraints, so i will approve it for now. I think the user would expect a redirect to a simple orders overview page, which shows all orders in a list. And (maybe) the same order overview page component as before placing the order + the deliver status.
> 
> Overall i like the look and feel how the order process is now implemented. Good job!

Thank You! Regarding the overview, you are absolutely right: It would be really nice but I would argue there is not enough time right now. I will resolve any conflicts regarding the branch and merge it into main afterwards.

**Pull request merged**